src/go/privileged-service/pkg/port/server.go

```diff
@@ -17,83 +17,93 @@ limitations under the License.
 package port
 
 import (
-	"errors"
+	"encoding/json"
 	"fmt"
-	"io"
 	"net"
 
+	"github.com/Microsoft/go-winio"
+	"github.com/pkg/errors"
 	"golang.org/x/sys/windows"
 	"golang.org/x/sys/windows/svc/debug"
+
+	"github.com/rancher-sandbox/rancher-desktop-agent/pkg/types"
 )
 
-const network = "tcp"
+const (
+	npipeEndpoint = "npipe:////./pipe/rancher_desktop/privileged_service"
+	protocol      = "npipe://"
+)
 
-// Server is a port server listening for port events over vtunnel
+// Server is a port server listening for port events from
+// RD Guest Agent over vtunnel.
 type Server struct {
-	host        string
-	port        int
 	eventLogger debug.Log
 	quit        chan interface{}
 	listener    net.Listener
 	stopped     bool
 }
 
-func NewServer(addr string, port int, elog debug.Log) *Server {
+// NewServer creates and returns a new instance of a Port Server.
+func NewServer(elog debug.Log) *Server {
 	return &Server{
-		host:        addr,
-		port:        port,
 		eventLogger: elog,
 		stopped:     true,
 	}
 }
 
 // Start initiates the port server on a given host:port
-// errCh is only used to write the initial error back to the caller
-func (s *Server) Start(errCh chan<- error) {
-	if !s.stopped {
-		return
-	}
+func (s *Server) Start() error {
 	s.quit = make(chan interface{})
-	l, err := net.ListenTCP(network, &net.TCPAddr{IP: net.ParseIP(s.host), Port: s.port})
+	c := winio.PipeConfig{
+		//
+		// SDDL encoded.
+		//
+		// (system = SECURITY_NT_AUTHORITY | SECURITY_LOCAL_SYSTEM_RID)
+		// owner: system
+		// ACE Type: (A) Access Allowed
+		// grant: (GA) GENERIC_ALL to (WD) Everyone
+		//
+		SecurityDescriptor: "O:SYD:(A;;GA;;;WD)",
```

Comment:
Note for future reference: Nino tested this, `IU` (interactive user) didn't seem to work.